cmd/helm/upgrade.go

```diff
@@ -247,23 +249,32 @@ func (u *upgradeCmd) run() error {
 		return err
 	}
 
+	// load the chart to update
+	chart, err := chartutil.Load(chartPath)
+	if err != nil {
+		return prettyError(err)
+	}
+
 	// Check chart requirements to make sure all dependencies are present in /charts
-	ch, err := chartutil.Load(chartPath)
-	if err == nil {
-		if req, err := chartutil.LoadRequirements(ch); err == nil {
-			if err := renderutil.CheckDependencies(ch, req); err != nil {
-				return err
-			}
-		} else if err != chartutil.ErrRequirementsNotFound {
-			return fmt.Errorf("cannot load requirements: %v", err)
+	if req, err := chartutil.LoadRequirements(chart); err == nil {
+		if err := renderutil.CheckDependencies(chart, req); err != nil {
+			return err
 		}
-	} else {
-		return prettyError(err)
+	} else if err != chartutil.ErrRequirementsNotFound {
+		return fmt.Errorf("cannot load requirements: %v", err)
+	}
+
+	if u.appVersion != "" {
+		if !chart.Metadata.OverrideAppVersion {
+			return fmt.Errorf("override app version is not allowed on this chart")
```

Comment:
Can you please improve the error message to say *what* the chart developer has to do to allow the app version to be overridden? We need a call-to-action so the user knows what they need to do to override the app version.

Reply:
I'd suggest something like:

> The chart maintainer has disallowed overriding the chart's package metadata. Please contact the maintainer to set `overrideChartMeta` in their Chart.yaml to `true`.